src/NLog/Targets/MailTarget.cs

```diff
@@ -372,24 +379,39 @@ private StringBuilder CreateBodyBuffer(IEnumerable<AsyncLogEventInfo> events, Lo
         }
 
         /// <summary>
-        /// Set propertes of <paramref name="client"/>
+        /// Set properties of <paramref name="client"/>
         /// </summary>
         /// <param name="lastEvent">last event for username/password</param>
         /// <param name="client">client to set properties on</param>
-        private void ConfigureMailClient(LogEventInfo lastEvent, ISmtpClient client)
+        internal void ConfigureMailClient(LogEventInfo lastEvent, ISmtpClient client)
         {
-
             CheckRequiredParameters();
 
-            var renderedSmtpServer = this.SmtpServer.Render(lastEvent);
-            if (string.IsNullOrEmpty(renderedSmtpServer))
+            if (this.SmtpServer == null && string.IsNullOrEmpty(this.PickupDirectoryLocation))
+            {
+                throw  new NLogRuntimeException(string.Format(RequiredPropertyIsEmptyFormat, "SmtpServer/PickupDirectoryLocation"));
+            }
+
+            var smtpServer = this.SmtpServer;
+            if (smtpServer != null)
             {
-                throw new NLogRuntimeException(string.Format(RequiredPropertyIsEmptyFormat, "SmtpServer"));
+                var renderedSmtpServer = smtpServer.Render(lastEvent);
+                if (string.IsNullOrEmpty(renderedSmtpServer) && string.IsNullOrEmpty(this.PickupDirectoryLocation))
+                {
+                
+                    throw new NLogRuntimeException(string.Format(RequiredPropertyIsEmptyFormat,
+                        string.IsNullOrEmpty(renderedSmtpServer) ? "SmtpServer" : "PickupDirectoryLocation"));
+                }
+                
+                client.Host = renderedSmtpServer;
+
+                // The network delivery method take precedence if both Host and PickupDirectoryLocation are present.
+                client.DeliveryMethod = SmtpDeliveryMethod.Network;
```

Comment:
@304NotModified should we also expose the [DeliveryMethod](https://msdn.microsoft.com/en-us/library/system.net.mail.smtpdeliverymethod(v=vs.110).aspx)  as part of the MailTarget properties, or can we assume  `Network` when there a Host and `SpecifiedPickupDirectory` where there not a Host but PickUpDirectoryValue? (should we also consider `PickupDirectoryFromIis`?)

regarding the other comments, will these kind of check be more appropriate ?